synapse/federation/transport/server.py

```diff
@@ -448,6 +448,50 @@ def _wrap(self, code):
         return code
 
 
+class OpenIdUserInfo(BaseFederationServlet):
+    """
+    Exchange a bearer token for information about a user.
+
+    The response format should be compatible with:
+        http://openid.net/specs/openid-connect-core-1_0.html#UserInfoResponse
+
+    GET /openid/userinfo?access_token=ABDEFGH HTTP/1.1
```

Comment:
Is `access_token` something part of the OpenID spec? Or defined by us? I'd rather not have two different meanings for `access_token`